indexing-hadoop/src/main/java/io/druid/indexer/path/DatasourcePathSpec.java

```diff
@@ -0,0 +1,186 @@
+  {
+    return maxSplitSize;
+  }
+
+  @Override
+  public Job addInputPaths(
+      HadoopDruidIndexerConfig config, Job job
+  ) throws IOException
+  {
+    final List<DataSegment> segments = indexerMetadataStorageCoordinator.getUsedSegmentsForInterval(
+        ingestionSpec.getDatasource(),
+        ingestionSpec.getInterval()
+    );
+    logger.info(
+        "Found total [%d] segments for [%s]  in interval [%s]",
+        segments.size(),
+        ingestionSpec.getDatasource(),
+        ingestionSpec.getInterval()
+    );
+
+    if (ingestionSpec.getDimensions() == null) {
+      List<String> dims;
+      if (config.getParser().getParseSpec().getDimensionsSpec().hasCustomDimensions()) {
+        dims = config.getParser().getParseSpec().getDimensionsSpec().getDimensions();
+      } else {
+        Set<String> dimSet = Sets.newHashSet(
+            Iterables.concat(
+                Iterables.transform(
+                    segments,
+                    new Function<DataSegment, Iterable<String>>()
+                    {
+                      @Override
+                      public Iterable<String> apply(DataSegment dataSegment)
+                      {
+                        return dataSegment.getDimensions();
+                      }
+                    }
+                )
+            )
+        );
+        dims = Lists.newArrayList(
+            Sets.difference(
+                dimSet,
+                config.getParser()
+                      .getParseSpec()
+                      .getDimensionsSpec()
+                      .getDimensionExclusions()
+            )
+        );
+      }
+      ingestionSpec = ingestionSpec.withDimensions(dims);
+    }
+
+    if (ingestionSpec.getMetrics() == null) {
+      Set<String> metrics = Sets.newHashSet();
+      AggregatorFactory[] cols = config.getSchema().getDataSchema().getAggregators();
+      for (AggregatorFactory col : cols) {
+        List<AggregatorFactory> deps = col.getRequiredColumns();
+        if (deps != null && deps.size() > 0) {
+          metrics.addAll(
+              Lists.transform(
+                  deps, new Function<AggregatorFactory, String>()
+                  {
+                    @Override
+                    public String apply(AggregatorFactory input)
+                    {
+                      return input.getName();
+                    }
+                  }
+              )
+          );
+        }
+      }
+      ingestionSpec = ingestionSpec.withMetrics(Lists.newArrayList(metrics));
+    }
+
+    job.getConfiguration().set(DatasourceInputFormat.CONF_DRUID_SCHEMA, mapper.writeValueAsString(ingestionSpec));
+    job.getConfiguration().set(DatasourceInputFormat.CONF_INPUT_SEGMENTS, mapper.writeValueAsString(segments));
+    job.getConfiguration().set(DatasourceInputFormat.CONF_MAX_SPLIT_SIZE, String.valueOf(maxSplitSize));
+    MultipleInputs.addInputPath(job, new Path("/dummy/tobe/ignored"), DatasourceInputFormat.class);
```

Comment:
since changing it has no impact and used only in once place. I don't see any advantage of truning it into a constant really.